clients/src/main/java/org/oracle/okafka/clients/admin/KafkaAdminClient.java

```diff
@@ -759,16 +791,25 @@ && handleUnsupportedVersionException((UnsupportedVersionException) throwable)) {
 			if (log.isDebugEnabled()) {
 				log.debug("{} failed: {}. Beginning retry #{}", this, prettyPrintException(throwable), tries);
 			}
-			runnable.enqueue(this, now);
+			maybeRetry(now, throwable);
 		}
+		
+		void maybeRetry(long now, Throwable throwable) {
```

Comment:
Have you tested this with any failure test ?

Reply:
This is actually working same as the previous logic, previously we were enqueuing again in case of an error but here we are calling a maybeRetry(). I did this because in kafka also they have changed this.